src/simulation.jl

```diff
@@ -66,14 +66,15 @@ function HorizontalConvectionSimulation(; Ra=1e11, h₀_frac=0.6, output_writer=
 
     underlying_grid = RectilinearGrid(
             architecture,
-            size = (Nx, Ny, Nz),
+            size = (Nx, Nz),
             x = (-Lx/2, Lx/2),
-            y = (-Ly/2, Ly/2),
+            #y = (-Ly/2, Ly/2),
             z = (-H, 0),
-            halo = (4,4,4),
-            topology = (Bounded, Periodic, Bounded))
+            halo = (4,4),
+            topology = (Bounded, Flat, Bounded))
```

Comment:
Since you'll want to switch back and forth between Flat and Periodic simulations, it's better for these to be conditional if statements than hard-coded changes and comments.

For example, you could do something like:
```julia
function HorizontalConvectionSimulation(; Ra=1e11, h₀_frac=0.6, Nx = 256, Ny=1, Nz = 32, output_writer=true, advection=true, architecture=CPU())

if Ny==1
    underlying_grid = RectilinearGrid(
            architecture,
            size = (Nx, Nz),
            x = (-Lx/2, Lx/2),
            z = (-H, 0),
            halo = (4,4),
            topology = (Bounded, Flat, Bounded))
else
    underlying_grid = RectilinearGrid(
            architecture,
            size = (Nx, Nz),
            x = (-Lx/2, Lx/2),
            y = (-Ly/2, Ly/2),
            z = (-H, 0),
            halo = (4,4,4),
            topology = (Bounded, Periodic, Bounded))
end
```

Should be self-explanatory how to generalize other parts of the script, e.g. the topography functions.